server/model/legal_hold.go

```diff
@@ -93,6 +93,14 @@ func (lh *LegalHold) IsValidForCreate() error {
 		return errors.New("LegalHold must end at a valid time or zero")
 	}
 
+	if lh.EndsAt > 0 && lh.StartsAt > lh.EndsAt {
+		return errors.New("LegalHold must end after it starts")
+	}
+
+	if !lh.ExcludePublicChannels && !mmConfigComplianceEnabled {
+		return errors.New("Compliance must be enabled on the Mattermost server in order to include public channels in a LegalHold")
```

Comment:
Let's be specific that this is Compliance Monitoring. The compliance section has a few different subsections.